lib/internal/streams/destroy.js

```diff
@@ -252,6 +252,25 @@ function constructNT(stream) {
       process.nextTick(emitConstructNT, stream);
     }
   });
+  if (result !== undefined && result !== null) {
+    try {
+      const then = result.then;
+      if (typeof then === 'function') {
+        then.call(result, undefined, function(err) {
+          if (r) {
+            r.constructed = true;
+          }
+          if (w) {
+            w.constructed = true;
+          }
+          called = true;
+          process.nextTick(errorOrDestroy, stream, err);
+        });
+      }
+    } catch (err) {
+      process.nextTick(emitErrorNT, stream, err);
```

Comment:
@jasnell FYI. You don't need to `nextTick` in stream impl. I think you've done this in a few places in quic as well.